modules/data/src/entity-services/entity-collection-service-base.ts

```diff
@@ -143,7 +143,7 @@ export class EntityCollectionServiceBase<
    */
   add(
     entity: Partial<T>,
-    options: EntityActionOptions & { isOptimistic: false }
+    options?: EntityActionOptions & { isOptimistic: false }
```

Comment:
Won't this change undo that desired behavior that @yharaskrik added? And make it worde because we allow a `Partia<T>` now?
I think the problem here, is that TS doesn't see this as an overload.
What about adding another overload where `options` is optional and T can't be a partial (the original implementation)?

```ts
  add(
    entity: Partial<T> | T,
    options: EntityActionOptions & { isOptimistic: false }
  ): Observable<T>;
  add(entity: T, options?: EntityActionOptions): Observable<T>;
  add(entity: T, options?: EntityActionOptions): Observable<T> {
    return this.dispatcher.add(entity, options);
  }
```

That way, you can either provide them, leave them out, and when optimistic is false then `T` is a partial.


![image](https://user-images.githubusercontent.com/28659384/122809495-77352680-d2ce-11eb-8923-e0e92e9bb282.png)

Reply:
You're right @timdeschryver I was mistakenly seeing the implementation as an overload. My only issue with the solution you suggested is that my expectation as a user would be that this would be acceptable:
```javascript 
this.storyDataService.add(story as Partial<Story>)
```
Given that isOtimistic: false is the default. But that's not the issue we're trying to solve here and is I believe the strict typing  @yharaskrik was aiming for and was accepted earlier so I'll update the PR.